Tests fail with packaging 26.0

With packaging 26.0, the tests fail as follows:

```
_______ test_evaluate_extra[platform_release >= '6'-environment10-True] ________

marker_string = "platform_release >= '6'"
environment = {'platform_release': '6.1-foobar'}, expected = True

    @pytest.mark.parametrize(
        ("marker_string", "environment", "expected"),
        [
            (f"os.name == '{os.name}'", None, True),
            ("sys.platform == 'win32'", {"sys_platform": "linux2"}, False),
            ("platform.version in 'Ubuntu'", {"platform_version": "#39"}, False),
            ("platform.machine=='x86_64'", {"platform_machine": "x86_64"}, True),
            (
                "platform.python_implementation=='Jython'",
                {"platform_python_implementation": "CPython"},
                False,
            ),
            (
                "python_version == '2.5' and platform.python_implementation!= 'Jython'",
                {"python_version": "2.7"},
                False,
            ),
            (
                (
                    "platform_machine in 'x86_64 X86_64 aarch64 AARCH64 ppc64le PPC64LE"
                    " amd64 AMD64 win32 WIN32'"
                ),
                {"platform_machine": "foo"},
                False,
            ),
            (
                (
                    "platform_machine in 'x86_64 X86_64 aarch64 AARCH64 ppc64le PPC64LE"
                    " amd64 AMD64 win32 WIN32'"
                ),
                {"platform_machine": "x86_64"},
                True,
            ),
            (
                (
                    "platform_machine not in 'x86_64 X86_64 aarch64 AARCH64 ppc64le PPC64LE"
                    " amd64 AMD64 win32 WIN32'"
                ),
                {"platform_machine": "foo"},
                True,
            ),
            (
                (
                    "platform_machine not in 'x86_64 X86_64 aarch64 AARCH64 ppc64le PPC64LE"
                    " amd64 AMD64 win32 WIN32'"
                ),
                {"platform_machine": "x86_64"},
                False,
            ),
            ("platform_release >= '6'", {"platform_release": "6.1-foobar"}, True),
            # extras
            # single extra
            ("extra != 'security'", {"extra": "quux"}, True),
            ("extra != 'security'", {"extra": "security"}, False),
            ("extra != 'security'", {}, True),
            ("extra != 'security'", {"platform_machine": "x86_64"}, True),
            # normalization
            ("extra == 'Security.1'", {"extra": "security-1"}, True),
            ("extra == 'a'", {}, False),
            ("extra != 'a'", {}, True),
            ("extra == 'a' and extra == 'b'", {}, False),
            ("extra == 'a' or extra == 'b'", {}, False),
            ("extra != 'a' and extra != 'b'", {}, True),
            ("extra != 'a' or extra != 'b'", {}, True),
            ("extra != 'a' and extra == 'b'", {}, False),
            ("extra != 'a' or extra == 'b'", {}, True),
            # multiple extras
            ("extra == 'a'", {"extra": ("a", "b")}, True),
            ("extra == 'a'", {"extra": ("b", "c")}, False),
            ("extra != 'a'", {"extra": ("a", "b")}, False),
            ("extra != 'a'", {"extra": ("b", "c")}, True),
            ("extra == 'a' and extra == 'b'", {"extra": ("a", "b", "c")}, True),
            ("extra == 'a' and extra == 'b'", {"extra": ("a", "c")}, False),
            ("extra == 'a' or extra == 'b'", {"extra": ("a", "c")}, True),
            ("extra == 'a' or extra == 'b'", {"extra": ("b", "c")}, True),
            ("extra == 'a' or extra == 'b'", {"extra": ("c", "d")}, False),
            ("extra != 'a' and extra != 'b'", {"extra": ("a", "c")}, False),
            ("extra != 'a' and extra != 'b'", {"extra": ("b", "c")}, False),
            ("extra != 'a' and extra != 'b'", {"extra": ("c", "d")}, True),
            ("extra != 'a' or extra != 'b'", {"extra": ("a", "b", "c")}, False),
            ("extra != 'a' or extra != 'b'", {"extra": ("a", "c")}, True),
            ("extra != 'a' or extra != 'b'", {"extra": ("b", "c")}, True),
            ("extra != 'a' and extra == 'b'", {"extra": ("a", "b")}, False),
            ("extra != 'a' and extra == 'b'", {"extra": ("b", "c")}, True),
            ("extra != 'a' and extra == 'b'", {"extra": ("c", "d")}, False),
            ("extra != 'a' or extra == 'b'", {"extra": ("a", "b")}, True),
            ("extra != 'a' or extra == 'b'", {"extra": ("c", "d")}, True),
            ("extra != 'a' or extra == 'b'", {"extra": ("a", "c")}, False),
        ],
    )
    def test_evaluate_extra(
        marker_string: str, environment: dict[str, str | set[str]] | None, expected: bool
    ) -> None:
        m = parse_marker(marker_string)

>       assert m.evaluate(environment) is expected
E       assert False is True
E        +  where False = <bound method SingleMarker.evaluate of <MarkerExpression platform_release >= "6">>({'platform_release': '6.1-foobar'})
E        +    where <bound method SingleMarker.evaluate of <MarkerExpression platform_release >= "6">> = <MarkerExpression platform_release >= "6">.evaluate

tests/marker/test_evaluation.py:151: AssertionError
__________________ test_arbitrary_unsupported[===abc->=1-and] __________________

a = '===abc', b = '>=1', operand = 'and'

    @pytest.mark.parametrize(
        "a, b, operand",
        [("===abc", ">=1", "and"), ("===1.0.0", "<1", "or"), ("===abc", "==1.*", "or")],
    )
    def test_arbitrary_unsupported(a: str, b: str, operand: str) -> None:
>       with pytest.raises(ValueError):
E       Failed: DID NOT RAISE <class 'ValueError'>

tests/specifier/test_arbitrary.py:43: Failed
```

--- a/src/dep_logic/markers/single.py
+++ b/src/dep_logic/markers/single.py
@@ -8,7 +8,7 @@ from dataclasses import dataclass, field, replace
 
 from packaging.markers import default_environment
 from packaging.specifiers import InvalidSpecifier, Specifier
-from packaging.version import InvalidVersion
+from packaging.version import InvalidVersion, Version
 
 from dep_logic.markers.any import AnyMarker
 from dep_logic.markers.base import BaseMarker, EvaluationContext
@@ -202,7 +202,7 @@ class MarkerExpression(SingleMarker):
                 pass
             else:
                 try:
-                    return spec.contains(lhs)
+                    return spec.contains(Version(lhs))
                 except InvalidVersion:
                     pass
 

--- a/src/dep_logic/specifiers/range.py
+++ b/src/dep_logic/specifiers/range.py
@@ -86,6 +86,8 @@ class RangeSpecifier(VersionSpecifier):
     def contains(
         self, version: UnparsedVersion, prereleases: bool | None = None
     ) -> bool:
+        if not isinstance(version, Version):
+            version = Version(version)
         return self.to_specifierset().contains(version, prereleases)
 
     def __invert__(self) -> BaseSpecifier:

